### PR TITLE
Create Project Redux Reworked---to make less number of files changed

### PR DESCRIPTION
**Pull request by 212040-AnnMariaJenson**

### CODE REVIEW

File: src/pages/compose/index.tsx [hunk 1]
```diff
@@ -37,18 +37,25 @@ const Compose = () => {
   const [descriptionError, setDescriptionError] = useState(false);
   const [deleteModalVisible, setDeleteModalVisible] = useState(false);
   const [projectId, setProjectId] = useState(1);
+  const [saveClicked, setSaveClicked] = useState(false);
+  const [savedData, setSavedData] = useState(null);
+  const dispatch = useDispatch();
+
+  //useSelector to take posted - compose page name and desc
+  const postComposeNameDescData = useSelector(postComposeNameDescSelector);
+  console.log("postComposeNameDescData",postComposeNameDescData)
```
Comment (sourcecoursepoc) [inline src/pages/compose/index.tsx]: console need to be removed

File: src/pages/compose/index.tsx [hunk 2]
```diff
@@ -58,23 +65,17 @@ const Compose = () => {
...
-      if (response.data !== -1) {
-        setProjectId(response.data.uid);
-        handleIconClick("HddFilled");
-      }
-    } catch (error) {
-      console.error(error);
+    //dispatch done to post compose page name and desc to db
+    const success = await dispatch(postComposeNameDescRequest(name, description));
+    if (success) {
+      showSuccessToast("Saved Successfully");
```
Comment (sourcecoursepoc) [inline src/pages/compose/index.tsx]: Messages and hardcoded values needs to be moved to constant file

File: src/redux/actions/composeAction.tsx [hunk 3]
```diff
@@ -196,4 +210,69 @@ export const searchSchemaByTagsInfoFailureAction = (
...
+      type: POST_COMPOSE_NAME_DESC_FAILURE,
+      payload: { error },
+  });
+  
+  
+  // GET COMPOSE NAME AND DESC
+  
+  export const getComposeNameDescRequest = (
+    // uid: any,
+    // name: any,
```
Comment (sourcecoursepoc) [inline src/redux/actions/composeAction.tsx]: Any purpose in keeping it? If not remove commented code

File: src/redux/reducers/composeReducer.tsx [hunk 4]
```diff
@@ -226,6 +231,46 @@ export const searchSchemaByTagReducer = (state = initialSearchSchemaState, actio
...
+  error: null,
+};
+
+export const postComposeNameDescReducer = (
+  state = initialPostNameDescState,
+  action: ComposeNameDescActions
+): ComposeNameDescState => {
+  switch (action.type) {
+    case POST_COMPOSE_NAME_DESC:
+      console.log('POST_COMPOSE_NAME_DESC action triggered');
```
Comment (sourcecoursepoc) [inline src/redux/reducers/composeReducer.tsx]: :(

Comment (sourcecoursepoc) [reply src/redux/reducers/composeReducer.tsx]: Remove

File: src/redux/sagas/composeSaga.tsx [hunk 5]
```diff
@@ -227,4 +231,28 @@ export function* searchSchemaByTagRequestSaga() {
...
+//saga for posting compose page name and description
+const postNameAndDescriptionAPI = BASE_URL + "/project";
+function postNameAndDescriptionAPIcall(name:any,description:any): Promise<AxiosResponse<any, any>> {
+  return axios.post(`${postNameAndDescriptionAPI}`,{ 
+  name:name,
+  description:description
+}
+  );
+}
+//console.log("postNameAndDescriptionAPIcall---saga",postNameAndDescriptionAPIcall())
```
Comment (sourcecoursepoc) [inline src/redux/sagas/composeSaga.tsx]: Remove commented parts

**Review by sourcecoursepoc — CHANGES_REQUESTED** [review]

File: src/pages/compose/index.tsx (hunk 1 above)
Comment (212040-AnnMariaJenson) [reply src/pages/compose/index.tsx]: Done

File: src/pages/compose/index.tsx (hunk 2 above)
Comment (212040-AnnMariaJenson) [reply src/pages/compose/index.tsx]: Done

File: src/redux/actions/composeAction.tsx (hunk 3 above)
Comment (212040-AnnMariaJenson) [reply src/redux/actions/composeAction.tsx]: Done

File: src/redux/reducers/composeReducer.tsx (hunk 4 above)
Comment (212040-AnnMariaJenson) [reply src/redux/reducers/composeReducer.tsx]: Done :)

File: src/redux/sagas/composeSaga.tsx (hunk 5 above)
Comment (212040-AnnMariaJenson) [reply src/redux/sagas/composeSaga.tsx]: Done

**Review by sourcecoursepoc — APPROVED** [review]

**Pull request merged**